### ISSUE TITLE
How to get current segment and it's playing time

### ISSUE THREAD

**Issue by piotr-cz** [videojs/videojs-contrib-hls#848]
## Description

I've prepared an external API to create picture snapshots from current video using FFmpeg (I've decided to not us client-side technology because the file has to be saved on server and I don't want to trust user input).

Key API parameters are _video URL_ and _time_ when to create a snapshot.
It doesn't make sense to pass index/ playlist file because at the time when API fetches it, it's generated new and so it's different to the one used by the player.

So I've decided to pass current segment URL and time at when to create a snapshot.

My current approach is that I'm starting a timer at the `loadedplaylist` event and when user pushes a button I'm counting the time that has passed since, retrieving the playlist media object and looping trough it's segments while substracting each segments duration.

``` js
// The time between button push and loadedplaylist event
var videoTime = (new Date() - this.playlistStartTimestamp) / 1e3;
var targetMedia = this.hls.playlists.media();

for (var i = 0, l = targetMedia.segments.length; i < l; i++) {
    videoSrc = targetMedia.segments[i].resolvedUri;

    if (videoTime < targetMedia.segments[i].duration) {
        break;
    }

    videoTime = videoTime - targetMedia.segments[i].duration;
}

// Send request to API
apiRequest({videoSrc: videoSrc, videoTime: videoTime});
```

The result is URL to current segment (`videoSrc`) and it's playing time (`videoTime`).

The problem is that result is very inaccurate, being off by few seconds in either direction when compared to what I see on screen.

Is there any way to retrieve current playing segment and it's playing time?
## Sources

Is a certain source or a certain segment affected? please provide a public (accesible over the internet) link to it below.

_The script is not public_
## Steps to reproduce

Explain in detail the exact steps necessary to reproduce the issue.
### Expected

Please describe what you expected to happen that did not happen in the description.
### Error output

If there are any errors in the console, from the player, or anywhere else please include them here:
## Additional Information

Please include any additional information necessary here. Including the following:
### videojs-contrib-hls version

what version of videojs-contrib-hls does this occur with?
_videojs-contrib-hls 3.5.3_
### videojs version

what version of videojs does this occur with?
_video.js 5.11.6_
### Browsers

what browsers are affected? please include browser and version for each
*
### Platforms

what platforms are affected? please include operating system and version or device and version for each
*
### Other Plugins

are any other videojs plugins being used on the page? If so, please list them with version below.
*
### Other JavaScript

are you using any other javascript libraries or frameworks on the page? if so please list them below.
*

**Comment by gesinger** [comment]
Sorry this ticket has gone a long time without much attention. We recently added the ability to track the actively playing segment via the segment-metadata text track: https://github.com/videojs/videojs-contrib-hls#segment-metadata . Please let us know if this doesn't resolve your use case. Thank you.

**Comment by piotr-cz** [comment]
Since which version is this available, is it [v5.3.3](https://github.com/videojs/videojs-contrib-hls/blob/master/CHANGELOG.md#533-2017-03-03)?

**Comment by piotr-cz** [comment]
ATM I'm using this simple code for playing stream. For paused stream I'm storing stale media and pausedAt timestamp.

```js
var targetMedia = this.hls.playlists.media();
var snapshotTime = this.player.currentTime();

var segment;
var segmentTime;

// Itinerate trough available segments and get first within which snapshotTime is
for (var i = 0, l = targetMedia.segments.length; i < l; i++) {
    // Note: segment.end may be undefined or is not properly set
    if (snapshotTime < targetMedia.segments[i].end) {
        segment = targetMedia.segments[i];
        break;
    }
}

// Null segmentTime in case it's lower then 0.
if (segment) {
    segmentTime = Math.max(0, snapshotTime - (segment.end - segment.duration));
// Because early segments don't have end property
} else {
    segment = targetMedia.segments[0];
    segmentTime = 0;
}

apiRequest({'videoTime': segmentTime, 'videoUri': segment.uri});
```

**Comment by mjneil** [comment]
@piotr-cz It is currently unreleased in master. I'm not sure when the next release will be but it should be soon

**Comment by piotr-cz** [comment]
ah, OK, I'll wait then - releases come pretty often.
I'd be cool if releases on [release page](https://github.com/videojs/videojs-contrib-hls/releases) would come with it's changelog segment like for example in [video.js releases](https://github.com/videojs/video.js/releases)

**Comment by gesinger** [comment]
Sorry about that @piotr-cz , and thanks @mjneil . It was just released as part of v5.4.0.
